ios/Capacitor/Capacitor/Plugins/CapacitorUrlRequest.swift

```diff
@@ -130,28 +118,27 @@ open class CapacitorUrlRequest: NSObject, URLSessionTaskDelegate {
 
                 data.append("\r\n".data(using: .utf8)!)
             } else if type == "string" {
-                data.append("\r\n--\(boundary)\r\n".data(using: .utf8)!)
+                data.append("--\(boundary)\r\n".data(using: .utf8)!)
                 data.append("Content-Disposition: form-data; name=\"\(key!)\"\r\n".data(using: .utf8)!)
                 data.append("\r\n".data(using: .utf8)!)
                 data.append(value.data(using: .utf8)!)
                 data.append("\r\n".data(using: .utf8)!)
             }
-
         }
-
-        data.append("\r\n--\(boundary)--\r\n".data(using: .utf8)!)
+        data.append("--\(boundary)--\r\n".data(using: .utf8)!)
 
         return data
     }
 
     public func getRequestData(_ body: JSValue, _ contentType: String, _ dataType: String? = nil) throws -> Data? {
+        let boundary = contentType.components(separatedBy: "=").last
         if dataType == "file" {
             guard let stringData = body as? String else {
                 throw CapacitorUrlRequestError.serializationError("[ data ] argument could not be parsed as string")
             }
             return Data(base64Encoded: stringData)
         } else if dataType == "formData" {
-            return try getRequestDataFromFormData(body)
+            return try getRequestDataFromFormData(body, boundary!)
```

Comment:
We sure this will always be non-nil?

Reply:
When using the patched fetch/XHR yeah, it will be there.
For direct API calls not so sure, but on Android there are no checks, so in any case it would crash on both platforms.

Reply:
Could easily turn this into:
```
} else if dataType == "formData", let boundary {
```
to avoid the force unwrapping

Reply:
another solution could be to collapse the `"multipart/form-data"` check below into one if like:
```swift
} else if dataType == "formData" || contentType.contains("multipart/form-data"), let let boundary = contentType.components(separatedBy: "=").last {
```
It's a bit verbose but it keeps the one form-data case in a single branch

Reply:
"formData" and "multipart/form-data" call different functions, so doing this would require a second if to call one or the other.